src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx

```diff
@@ -0,0 +1,129 @@
+
+import { PanelTeam } from './PanelTeam';
+
+type DictTeams = {
+  [page: string]: TeamResponse[];
+};
+
+export const PanelTeams = () => {
+  const { t } = useTranslation();
+
+  const [dictTeams, setDictTeams] = useState<DictTeams>({});
+  const [page, setPage] = useState(1);
+  const ordering = useTeamStore((state) => state.ordering);
+  const { data, isError, isLoading } = useTeams({
+    page,
+    ordering,
+  });
+  const containerRef = useRef<HTMLDivElement>(null);
+  const scrollLoading = useRef(false);
+  const teams = useMemo(() => {
+    let teams: TeamResponse[] = [];
+    Object.keys(dictTeams)
+      .sort()
+      .forEach((key) => (teams = teams.concat(dictTeams[key])));
+
+    return teams;
+  }, [dictTeams]);
+
+  useEffect(() => {
+    setDictTeams({});
+    setPage(1);
+    setTimeout(() => {
+      if (!containerRef.current) {
+        return;
+      }
+
+      containerRef.current.scroll({
+        top: 0,
+        behavior: 'smooth',
+      });
+    }, 50);
+  }, [ordering]);
+
+  useEffect(() => {
+    if (!data?.results) {
+      return;
+    }
+
+    setDictTeams((prev) => {
+      const newTeams = {
+        [page]: data.results,
+      };
+
+      scrollLoading.current = false;
+
+      return {
+        ...prev,
+        ...newTeams,
+      };
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data?.results]);
+
+  return (
+    <Box $height="90vh" $css="overflow: auto;" ref={containerRef}>
+      <InfiniteScroll
+        hasMore={data?.count ? data.count > teams.length : false}
+        isLoading={scrollLoading.current}
+        next={() => {
+          scrollLoading.current = true;
+          setPage((page) => page + 1);
+        }}
+        scrollContainer={containerRef.current}
+        $gap="1rem"
+        as="ul"
+        className="p-s mt-t"
+        role="listbox"
+      >
+        {isLoading && (
+          <Box $align="center" className="m-l">
+            <Loader />
+          </Box>
+        )}
+
+        {data && !data.count && (
+          <Box $justify="center">
+            <Text
+              as="p"
+              className="mb-0 mt-0"
+              $theme="greyscale"
+              $variation="500"
+            >
+              {t('0 group to display.')}
+            </Text>
+            <Text as="p" $theme="greyscale" $variation="500">
+              {t(
+                'Create your first team by clicking on the "Create a new team" button.',
+              )}
+            </Text>
+          </Box>
+        )}
+
+        {isError && (
+          <Box $justify="center" className="mb-b">
+            <Text $theme="danger" $align="center" $textAlign="center">
+              {t('Something bad happens, please refresh the page.')}
+            </Text>
+          </Box>
+        )}
+
+        {!isError &&
+          teams.map((team) => (
+            <Fragment key={team.id}>
+              <PanelTeam team={team} />
+            </Fragment>
+          ))}
```

Comment:
As, `isLoading`, `data && !data.count` and `isError` are not exclusive, these 3 blocks could be displayed altogether?

Reply:
https://github.com/numerique-gouv/people/pull/31/commits/ccef7d82b26ff1a60c90314be58c89d48aa38580